ovmf/tdx_data.go

```diff
@@ -138,17 +151,19 @@ func validateTDXMetadataSections(firmwareLen uint32, rawMetadata *abi.TDXMetadat
 
 type tdxFwParser struct {
 	Regions            []*MaterialGuestPhysicalRegion
+	PermRegions        []*MaterialGuestPhysicalRegion
 	Sections           []*abi.TDXMetadataSection
 	TDHOBregion        *MaterialGuestPhysicalRegion
 	DisableEarlyAccept bool
 	// MeasureAllRegions forces all regions to be measured, even if they are not marked as
 	// extendable in the metadata. This is only to be compatible with earlier versions
 	// Google's hypervisor.
-	MeasureAllRegions bool
+	MeasureAllRegions   bool
+	AllowTDShimMetadata bool
 }
 
 func (p *tdxFwParser) validateMetadataSectionGpr(sectionType uint32, gpr GuestPhysicalRegion) error {
-	for _, region := range p.Regions {
+	for _, region := range append(p.Regions, p.PermRegions...) {
```

Comment:
I don't think you want to be modifying p.Regions here, but append is allowed to modify it. Make a copy of it.

Reply:
Done.